lib/engine/step/tokener.rb

```diff
@@ -55,6 +55,11 @@ def place_token(entity, city, token, connected: true, extra_action: false,
                            "#{special_ability.city}, not on city #{city.index}"
         end
 
+        if special_ability&.type == :token && special_ability.hexes.any? && !special_ability.hexes.include?(hex.id)
```

Comment:
Only do the check if `!@game.loading`, so that we don't recheck every time the game loads, slowing down the load speed.

Reply:
would this check wrap the two other city and hex checks in this function?

Reply:
What do you mean?

Reply:
should the game-loading check wrap all the code from line 52 to ~72?
(the various sanity checks for city/hex)

Reply:
Yes, good catch.